projects/vanilla/src/app/search/search-configurator.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, OnDestroy, inject } from "@angular/core";
+import { MetadataConfig } from "@sinequa/components/metadata";
+import { SearchService } from "@sinequa/components/search";
+import { UIService } from "@sinequa/components/utils";
+import { AppService } from "@sinequa/core/app-utils";
+import { LoginService } from "@sinequa/core/login";
+import { Record, Results } from "@sinequa/core/web-services";
+import { Subscription } from "rxjs";
+import { METADATA_CONFIG } from "../../config";
+
+@Component({
+  selector: 'app-search-configurator',
+  templateUrl: './search-configurator.component.html'
+})
+export class SearchConfiguratorComponent implements OnDestroy {
+  /**
+   * Returns the configuration of the metadata displayed in the facet-preview component.
+   * The configuration from the config.ts file can be overridden by configuration from
+   * the app configuration on the server
+   */
+  public get metadata(): string[] {
+    const m = this.appService.app?.data?.metadata as any as MetadataConfig[] || METADATA_CONFIG;
+    return m.map(i => i.field);
+  }

```

Comment:
It makes sense to manage configuration like this in Vanilla Search.
But in Vanilla Builder, the configuration should always be associated to the components configuration. So there should not be a `METADATA_CONFIG` object. It should go directly in the ui-builder component's configuration object.